server/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeFiltersTests.java

```diff
@@ -295,6 +295,26 @@ public void testOnlyAttributeValueFilter() {
         assertThat(discoveryNodeFilters.isOnlyAttributeValueFilter(), is(discoveryNodeFilters.match(node)));
     }
 
+    public void testNormalizesIPAddressFilters() {
+        Settings settings = shuffleSettings(Settings.builder()
+            .put("xxx." + randomFrom("_ip", "_host_ip", "_publish_ip"), "fdbd:dc00:111:222:0:0:0:333")
+            .build());
+        DiscoveryNodeFilters filters = buildFromSettings(OR, "xxx.", settings);
+
+        DiscoveryNode node = new DiscoveryNode("", "", "", "", "fdbd:dc00:111:222::333", localAddress, emptyMap(), emptySet(), null);
+        assertThat(filters.match(node), equalTo(true));
+    }
+
+    public void testHostnameGetMatchedAndNotAffectedByNormalizing() {
+        Settings settings = shuffleSettings(Settings.builder()
+            .put("xxx._host", "test-host")
+            .build());
+        DiscoveryNodeFilters filters = buildFromSettings(OR, "xxx.", settings);
+
+        DiscoveryNode node = new DiscoveryNode("", "", "", "test-host", "192.168.0.1", localAddress, emptyMap(), emptySet(), null);
+        assertThat(filters.match(node), equalTo(true));
```

Comment:
Sorry I think I wasn't clear. What I meant was if, say, the node name is `fdbd:dc00:111:222::333` then this should not match the filter `xxx._name: fdbd:dc00:111:222:0:0:0:333` (and vice versa). In other words, a test that fails on 48132af26edcc317f8e71e6dc98f6a4277011fc2 to demonstrate why this isn't the right implementation.